backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/PostgresExecutor.java

```diff
@@ -125,16 +126,26 @@ public Mono<Void> executeVoid(Function<DSLContext, Mono<?>> queryFunction) {
     }
 
     public Flux<Record> executeRows(Function<DSLContext, Flux<Record>> queryFunction) {
+        return executeRows(queryFunction, !EAGER_FETCH);
+    }
+
+    public Flux<Record> executeRows(Function<DSLContext, Flux<Record>> queryFunction, boolean isEagerFetch) {
```

Comment:
Document in a comment what `isEagerFetch` is doing, why it is needed, when it is Ok to use, when it is not Ok to use, and which are the alternative I can use.